app/main.py

```diff
@@ -1,2 +1,66 @@
+from typing import Hashable, Any
+
+
 class Dictionary:
-    pass
+
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.threshold = int(self.capacity * 2 / 3)
+        self.length = 0
+        self.hash_table = [[] for _ in range(self.capacity)]
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        key_hash = hash(key)
+        index = key_hash % self.capacity
+        while True:
+            if len(self.hash_table[index]) == 0:
+                self.hash_table[index] = [key, value, key_hash]
+                self.length += 1
+                break
+            if self.hash_table[index][0] == key and \
```

Comment:
it's better to compare hashes first. comparing hashes that are represented by integers is faster than comparing complex keys (e.x. tuples)